apps/namadillo/src/hooks/useTransactionCallbacks.tsx

```diff
@@ -1,29 +1,36 @@
-import { accountBalanceAtom } from "atoms/accounts";
+import { accountBalanceAtom, defaultAccountAtom } from "atoms/accounts";
 import { shouldUpdateBalanceAtom, shouldUpdateProposalAtom } from "atoms/etc";
-import { claimableRewardsAtom } from "atoms/staking";
+import { claimableRewardsAtom, clearClaimRewards } from "atoms/staking";
 import { useAtomValue, useSetAtom } from "jotai";
 import { useTransactionEventListener } from "utils";
 
 export const useTransactionCallback = (): void => {
   const { refetch: refetchBalances } = useAtomValue(accountBalanceAtom);
   const { refetch: refetchRewards } = useAtomValue(claimableRewardsAtom);
+  const { data: account } = useAtomValue(defaultAccountAtom);
   const shouldUpdateBalance = useSetAtom(shouldUpdateBalanceAtom);
 
   const onBalanceUpdate = (): void => {
     // TODO: refactor this after event subscription is enabled on indexer
     shouldUpdateBalance(true);
     refetchBalances();
-    refetchRewards();
 
     const timePolling = 6 * 1000;
     setTimeout(() => shouldUpdateBalance(false), timePolling);
+
+    if (account?.address) {
+      clearClaimRewards(account.address);
+      setTimeout(() => refetchRewards(), timePolling);
+    }
   };
 
   useTransactionEventListener("Bond.Success", onBalanceUpdate);
   useTransactionEventListener("Unbond.Success", onBalanceUpdate);
   useTransactionEventListener("Withdraw.Success", onBalanceUpdate);
   useTransactionEventListener("Redelegate.Success", onBalanceUpdate);
-  useTransactionEventListener("ClaimRewards.Success", onBalanceUpdate);
+  useTransactionEventListener("ClaimRewards.Success", onBalanceUpdate, [
+    account,
```

Comment:
nice idea! I'm asking myself if this this object changes for whatever reason and if we should use `account?.address` instead